language/stdlib/compiled/src/transaction_scripts.rs

```diff
@@ -49,6 +49,7 @@ pub enum StdlibScript {
     RotateDualAttestationInfo,
     RotateSharedEd2551PublicKey,
     UpdateAccountLimitWindowInfo,
+    SetOpenScript,
```

Comment:
We can probably omit a script for this, since it's something we'll do only once. I think a good rule of thumb is: if it's a LR-only operation that we will only do once, no need for a script.

Reply:
The problem is that we have a number of locations in the codebase that uses this script. If we don't use this approach then we need to have those users to import move compiler to compile the scripts, which I'm not sure how easy would that be.

Reply:
Ah, I see that. Maybe we could export a constant for the script binary (similar to what we do [here](https://github.com/libra/libra/blob/master/language/e2e-tests/src/common_transactions.rs#L17)?). I know this is somewhat of a pain, but I am very hesitant to clutter the public allowlist/API docs/generated script bindings with test-only scripts. I have been on a personal crusade to make `StdlibScript` as small as possible by refactoring to delete other test-only scripts...

Reply:
Would adding `pub`'s to these constants be workable for this? https://github.com/libra/libra/blob/master/language/transaction-builder/generated/src/stdlib.rs#L1744?

One thing we will want to do after this PR is separate out "admin" scripts that actually need/should be on the allowlist out, and remove those that no longer need to be there.

Reply:
> One thing we will want to do after this PR is separate out "admin" scripts that actually need/should be on the allowlist out, and remove those that no longer need to be there.

I think this will address a lot of my concerns (e.g., it's perfect for `create_parent_vasp`). But I'm still not sure if we want the `set_open_script` in the admin script bindings because it's a special operation that we'll only do once at a point far in the future. Thoughts on the "will only do it once" -> "don't clutter the bindings with it" rule of thumb?

Reply:
I agree that `set_open_script` doesn't need to be included in the admin scripts. The proposal I have in my head right now is something like:
```
transaction_scripts/
  |-> public/ # This contains public scripts that need to be in the allowlist (e.g., create_parent_vasp)
  |-> testing/ # Contains scripts used for testing only (both admin/non-admin)
```
Only the `public` scripts would be included in the API/external bindings. Exactly how we want to expose the `testing` scripts is a question that I'm still thinking through, but I think we have a number of decent options on it. 

The specific LR, TC, and admin scripts needed by AOS would then be separated out from this, and we should see about moving the building of those scripts to the AOS side of things if possible since that will be needed in the future I'm sure, would declutter the API, and hopefully make things simpler :) 

cc @moeziniaf to get his thoughts on this idea.

Reply:
The `testing` idea seems great. It addresses all of my concerns and is also much nicer than the "constant that holds a binary" solution.

Reply:
Tim and I discussed offline exposing transaction_scripts in a more comprehensible way. I'm not sure exactly which scripts would fall under 'testing', but clearly any/all scripts which are for single client use (AOS) should be decoupled from the rest. (The rest could be broken down by Role of invocator).

IMO it doesn't even need to be 'only build these scripts from AOS',  since there is an argument to be made that the semantics of building move scripts and the modules they call should be tied to the language/ dir, but at least a logical separation between public/ vs aos/ scripts should be made.

Having a public API for flipping a boolean (once) seems like unnecessary clutter to me, but the question rides on the frequency of these one-off (non writeset) scripts and how we want to handle them. It seems that it could be useful to have these scripts saved somewhere (even on AOS) and then have an 'execute admin script X'  (not necessarily on behalf of another account?) where script X is operation data.